docs/DevGuide/Parallelization.md

```diff
@@ -564,6 +571,103 @@ calls on all array elements occurred in the same order. It is **undefined**
 behavior if the contribute calls are made in different orders on different array
 elements.
 
+# Mutable items in the GlobalCache
+
+Most items in the GlobalCache are constant, and are determined at
+initialization by type aliases called `const_global_cache_tags` as
+described above. However, the GlobalCache can also store mutable
+items. Because of asynchronous execution, care must be taken when
+mutating items in the GlobalCache, as described below.
+
+A mutable item can be of any type, as long as that type is something
+that can be checked for whether it is "up to date".  Here "up to date"
+means that the item can be safely used (even read-only) without
+needing to be mutated first. For example, a mutable item might be a
+function of time that knows the range of times for which it is valid;
+the mutable item is then deemed up to date if it will be called for a
+time within its range of validity, and it is deemed not up to date if
+it will be called for a time outside its range of validity.  Thus the
+up-to-date status of a mutable item is determined by both the state of
+the item itself and by the code that wishes to use that item.
+
+## 1. Initialization of mutable GlobalCache items
+
+Mutable GlobalCache items are initialized by specifying them via a
+type alias `mutable_global_cache_tags`, which is treated the same way
+as `const_global_cache_tags` for const items.
+
+## 2. Use of mutable GlobalCache items
+
+### 1. Checking if the item is up to date
+
+Because execution is asynchronous, any code that uses a mutable item
+in the GlobalCache must first check whether that item is up to date.
+The information about whether an item is up to date is assumed to be
+stored in the item itself.  For example, a mutable object stored in
+the GlobalCache might have type `std::map<temporal_id,T>` (for some
+type `T`), and then any code that uses the stored object can check
+whether an entry exists for a particular `temporal_id`.  To avoid
+race conditions, it is
+important that up-to-date checks are based on something that is
+independent of the order of mutation (like a `temporal_id`, and not
+like checking the size of a vector).
+
+To check an item, use the function
+`Parallel::mutable_cache_item_is_ready`, which returns a bool
+indicating whether the item is up to date.  If the item is up to date,
+then it can be used.  `Parallel::mutable_cache_item_is_ready` takes a
+lambda as an argument.  This lambda is passed a single argument: a const
+reference to the item being retrieved.  The lambda should determine
+whether the item is up to date. If so, it should return boost::none;
+if not, it should return a boost::optional<CkCallback> to a callback
```

Comment:
please put ticks around things like `optional`  and `boost::none` (and also use std::optional instead of boost:optional)